### ISSUE TITLE
Not able to to assign custom roles to Multi Cluster Apps.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#18326]
Rancher Server version - Build from master

Steps to reproduce the problem:
Log in to rancher server  , as a regular user who has access to it.
Create a cluster.

scenario 1:
Deploy a Multi cluster app in "Default" project with role set to "Manage Ingress" and "Manage workloads"

 Multi cluster app deployment fails with following error:
```user u-4ntonnmxa5 does not have all project roles provided in project p-8pvxm```

Expected Behavior:
Creating  MultiClusterApp should succeed.

**Comment by mrajashree** [comment]
I can not reproduce scenario 2, I tried these steps:
1. Create user and add to Default project with custom roles Manage Ingress and Manage Workloads
2. Login as this user and create Multiclusterapp with only one target project, the above default project, and pass roles Manage Ingress and Manage Workloads.
3. Multiclusterapp got created without errors

**Comment by mrajashree** [comment]
@sangeethah Behavior for scenario 1 will not be changed for 2.2

**Comment by sangeethah** [comment]
@mrajashree If we are not planning to the change this behavior , we should have the UI present only the list of roles that the creator has in common across all the targets.

The current API error message does not convey enough for user to take any corrective action. If we do not want the UI to be fixed , then we should have the error message presented to the user clearly state what roles are missing from the target project and suggest corrective measures.

cc @cjellick @alena1108

**Comment by vincent99** [comment]
Yes the UI can disable the role template choices that do not correspond to a PRTB/CRTB that the current (non-admin) user has.

**Comment by cjellick** [comment]
Agree with Vincent. Let's do that. Offline, we discussed going one step further such that the only roles shown in the drop down are the roles that the user has across all the target projects. But I think that is taking it too far and the logic is going to be too complicated. So, lets just limit the roles listed to roles the user has (regardless of whether they match all projects).

As Vince suggests, we'll also improve the error message on the backend to make it more clear why it fails

**Comment by mrajashree** [comment]
This PR changes the error message to include the roles that user is missing in each target https://github.com/rancher/rancher/pull/18488/files

**Comment by mrajashree** [comment]
@westlywright Backend for changing the api error message is merged. Do you want me to open a separate issue to track the UI changes?

**Comment by mrajashree** [comment]
@westlywright Backend changes are merged

**Comment by westlywright** [comment]
as discussed with @cjellick @alena1108 @mrajashree for 2.2 release the UI will now only present two options for mc app roles. `project-member` and `cluster-owner` with clearer label text describing exactly what these roles are doing.

**Comment by cjellick** [comment]
@westlywright for the case where the app is created via the API and uses a different role, can we have a read-only display section that just says "other roles"?
I think we can ship 2.2.0 without this, but it would be a nice to have

**Comment by westlywright** [comment]
@cjellick the ui changes are in, should I move this ticket or are we waiting on additional changes?

**Comment by cjellick** [comment]
Related to https://github.com/rancher/rancher/issues/18521, which encompasses the backend changes needed

**Comment by sangeethah** [comment]
Rancher server build from master - version c8da22fb

Check made From UI:

1. Multicluster App can be assigned to only 1 of the 2 roles - "Project member" (Default)/ "Cluster Owner".  Roles is of "radio button" type and forces one of the  roles to be selected. 

Check made for API:
1. As global admin user , create a cluster - "mytest" .
2. Add user - "test1" to "Default" project as "read-only" member.
3. Log in as test1. 
4. Create a MCAPP with target project as  "Default" from "mytest" cluster.

MCAPP creation fails with following error message - 
"user u-agr5z74r34 does not have following roles in project Default: [project-member]"
 
This error message should  be improved to include cluster name ( right now only Project name is provided) and the User Id can be removed (or add name instead of user id) .

**Comment by mrajashree** [comment]
Okay, will add cluster name along with the project name, will remove userId since it seems unnecessary

**Comment by sangeethah** [comment]
Tested with latest build from master:

Following error message is presented to the user which also has cluster information:	
"user does not have following roles in project hello2 of cluster santestuser2cluster1: [project-member]"

Note - When there are more than 1 project that does not have correct roles , API error provides information about only the one of them.
